src/mono/wasm/runtime/pthreads/shared/index.ts

```diff
@@ -132,16 +133,19 @@ export function isMonoWorkerMessagePreload<TPort>(message: MonoWorkerMessage<TPo
 }
 
 let synchronization_context_installed = false;
-export function install_synchronization_context(): void {
-    if (MonoWasmThreads && !synchronization_context_installed) {
-        runtimeHelpers.javaScriptExports.install_synchronization_context();
-        synchronization_context_installed = true;
+export function install_synchronization_context(current_thread_id: number): void {
+    if (MonoWasmThreads) {
+        if (!synchronization_context_installed) {
+            synchronization_context_installed = true;
+            runtimeHelpers.javaScriptExports.install_synchronization_context(current_thread_id);
+            mono_log_debug(`Installed JSSynchronizationContext to thread 0x${current_thread_id.toString(16)}`);
+        }
     }
 }
 
 export function assert_synchronization_context(): void {
     if (MonoWasmThreads) {
-        // TODO mono_assert(synchronization_context_installed, "Synchronization context not installed on the current worker. Please use dedicated worker for working with JavaScript interop.");
+        mono_assert(synchronization_context_installed, "Synchronization context not installed on the current worker. Please use dedicated worker for working with JavaScript interop.");
```

Comment:
this may need to point towards documentation or at least the name of the API you use to create a dedicated worker. i would have no idea how to resolve this assertion